src/onion_messenger.rs

```diff
@@ -299,6 +302,65 @@ async fn lookup_onion_support(pubkey: &PublicKey, client: &mut tonic_lnd::Lightn
     }
 }
 
+/// connect_to_onion_peers picks a few random nodes from the network graph that support onion messenging, then
+/// connects to them.
+pub(crate) async fn connect_to_onion_peers(
+    mut connector: impl PeerConnector + std::marker::Send + 'static,
+    target_peers: u8,
+    current_peers: HashMap<PublicKey, bool>,
+    lnd_key: PublicKey,
+) -> Result<(), Status> {
+    let graph = connector.describe_graph().await?;
+
+    let onion_peers = get_onion_peers(graph, target_peers, current_peers, lnd_key).await;
+    //for i in 0..onion_peers.len() {
+    for peer in onion_peers.iter() {
+        connector
+            .connect_peer(peer.0.clone(), peer.1.clone())
+            .await?
```

Comment:
Does the connect peer API return an error if LND fails to connect to a peer?
In my testing, LNDK doesn't print error logs on connection failure.